idaes/models/costing/SSLW.py

```diff
@@ -1016,13 +1016,32 @@ def cost_compressor(
             integer: whether the number of units should be constrained to be
                 an integer or not (default = True).
         """
-        # Confirm that unit is a turbine
+        # Confirm that unit is a compressor
         if not blk.unit_model.config.compressor:
             raise TypeError(
                 "cost_compressor method is only appropriate for "
                 "pressure changers with the compressor argument "
                 "equal to True."
             )
```

Comment:
If you call the `cost_compressor` function on a unit model that's not a compressor, the user deserves what they get.

Reply:
It might be best to log a `warning` in these cases. I agree there are situations where you want to use a different unit model with the compressor costing.